src/app/cli/src/explore/api_server.rs

```diff
@@ -26,14 +30,16 @@ pub struct APIServer {
     task_executor: Arc<dyn TaskExecutor>,
     flow_service: Arc<dyn FlowService>,
     time_source: Arc<dyn SystemTimeSource>,
+    access_token: String,
 }
 
 impl APIServer {
-    pub fn new(
+    pub async fn new(
```

Comment:
1) Can we avoid async new? It's not like a serious rule, but usually constructors should be sync and should not return errors. 

2) Perhaps, the token generation logic might be moved to command level. `APIServer` does not use it anyhow, it just generates it and provides a getter, but no other usage. This is an indication it's likely not owning this responsibility.

Reply:
Noticed for the future as well